### DISCUSSION TITLE
After updating docker image from .NET 6 to .NET 8 odbcinst can't be found after unixodbc installation.

### DISCUSSION THREAD

**Discussion by ruvinskyan** [dotnet/dotnet-docker#5023] [Q&A]
After updating docker image to .NET 8 I can't run "odbcinst -j" command, even though unixodbc installed based on a log.

My Dockerfile:
```
FROM mcr.microsoft.com/dotnet/aspnet:8.0 AS base

RUN apt-get update && apt-get install --no-install-recommends -y \
    curl \
    unixodbc \
    unixodbc-dev \
    && rm -rf /var/lib/apt/lists/* 
    
ENV PATH=/opt/sqlanywhere16/bin64:$PATH
ENV ODBCINI=/etc/odbc.ini
ENV ODBC_INI=/etc/odbc.ini
    
# Configure ODBC
WORKDIR /etc
RUN printf "[ODBC]\nTraceFile=/tmp/sql.log\nTrace=Yes" > odbcinst.ini
RUN printf "[SQL Anywhere 16]\nDescription=SAP SQL Anywhere 16 Driver\nDriver=/opt/sqlanywhere16/lib64/libdbodbc16_r.so\nSetup=/opt/sqlanywhere16/lib64/libdbodbc16_r.so" > template
RUN odbcinst -i -d -f template
```

It is failing on last line when trying to run odbcinst.

I ran container and tried to execute this command by my self from container but received same error. 
![image](https://github.com/dotnet/dotnet-docker/assets/46845518/092ecbe3-d893-4f1c-a477-e3d12b6a346a)

Also I tried to install unixodbc from container but it says it is already installed:
![image](https://github.com/dotnet/dotnet-docker/assets/46845518/9cc87875-d326-4d7d-ba28-0cda0f9a1e10)

Maybe somebody knows what can be a problem?

Thank you!

**Comment by lbussell** [comment]
This appears to be a result of upgrading from Debian 11 (Bullseye) to Debian 12 (Bookworm) and is unrelated to any changes in the .NET layers of the image.

In Bullseye:
```
> docker run --rm debian:bullseye-slim /bin/bash -c "apt-get -qq update && apt-get install -y unixodbc && odbcinst -j"
...
Works fine
```

In Bookworm:
```
> docker run --rm debian:bookworm-slim /bin/bash -c "apt-get -qq update && apt-get install -y unixodbc && odbcinst -j"
...
Doesn't work.
```

However, I did get it to work by directly installing the `odbcinst` command directly:
```
> docker run --rm debian:bookworm-slim /bin/bash -c "apt-get -qq update && apt-get install -y unixodbc && odbcinst -j"
...
The following additional packages will be installed:
  libltdl7 libodbc2 libodbcinst2 libreadline8 readline-common unixodbc-common
Suggested packages:
  odbc-postgresql tdsodbc readline-doc
The following NEW packages will be installed:
  libltdl7 libodbc2 libodbcinst2 libreadline8 odbcinst readline-common
  unixodbc unixodbc-common
0 upgraded, 8 newly installed, 0 to remove and 0 not upgraded.
...
unixODBC 2.3.11
DRIVERS............: /etc/odbcinst.ini
SYSTEM DATA SOURCES: /etc/odbc.ini
FILE DATA SOURCES..: /etc/ODBCDataSources
USER DATA SOURCES..: /root/.odbc.ini
SQLULEN Size.......: 8
SQLLEN Size........: 8
SQLSETPOSIROW Size.: 8
```

**Comment by AdenHous** [comment]
When I try this I get the error below, works fine with bulleye-slim. Docker build server is RHEL server.
sudo docker run --rm debian:bookworm-slim /bin/bash -c "apt-get -qq update && apt-get install -y libc6-dev && libgdiplus && libx11-dev -j"
[sudo] password for aadebayo:
Unable to find image 'debian:bookworm-slim' locally
Trying to pull repository registry.access.redhat.com/debian ...
Pulling repository registry.access.redhat.com/debian
Trying to pull repository registry.redhat.io/debian ...
Trying to pull repository docker.io/library/debian ...
bookworm-slim: Pulling from docker.io/library/debian
09f376ebb190: Pull complete
Digest: sha256:804194b909ef23fb995d9412c9378fb3505fe2427b70f3cc425339e48a828fca
Status: Downloaded newer image for docker.io/debian:bookworm-slim
W: GPG error: http://deb.debian.org/debian bookworm InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 0E98404D386FA1D9 NO_PUBKEY 6ED0E7B82643E131 NO_PUBKEY F8D2585B8783D481
E: The repository 'http://deb.debian.org/debian bookworm InRelease' is not signed.
W: GPG error: http://deb.debian.org/debian bookworm-updates InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 0E98404D386FA1D9 NO_PUBKEY 6ED0E7B82643E131
E: The repository 'http://deb.debian.org/debian bookworm-updates InRelease' is not signed.
W: GPG error: http://deb.debian.org/debian-security bookworm-security InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY 54404762BBB6E853 NO_PUBKEY BDE6D2B9216EC7A8
E: The repository 'http://deb.debian.org/debian-security bookworm-security InRelease' is not signed.
E: Problem executing scripts APT::Update::Post-Invoke 'rm -f /var/cache/apt/archives/*.deb /var/cache/apt/archives/partial/*.deb /var/cache/apt/*.bin || true'
E: Sub-process returned an error code

Any help will be appreciated.

**Reply by AdenHous** [reply]
Found out error was due to an older version of docker daemon running on the server.

**Reply by shawndutton-unitas** [reply]
@AdenHous , thank you for posting this.  I had the same error and this really helped me.  I also had to update my docker daemon in order to resolve this.
